scanner/templates/node/Dockerfile

```diff
@@ -5,17 +5,22 @@ ARG NODE_VERSION={{ .nodeVersion }}
 {{ if .yarn -}}
 ARG YARN_VERSION={{ .yarnVersion }}
 {{ end }}
-RUN apt-get update; apt install -y curl python-is-python3 pkg-config build-essential && \
+RUN apt-get update; apt install -y curl python-is-python3 pkg-config build-essential {{ if .prisma }}openssl {{ end }}&& \
```

Comment:
Do we _really_ need `curl`, `python-is-python3`, `pkg-config` & `build-essential`?

We didn't use them in the `remix`/`remix_prisma` templates & I'd like to have the `Dockerfile` as clean as possible, like we had in the other templates tbh.
Especially since we will have these files (`Dockerfile` & `start_with_migrations`) in the source code of our [Remix Stacks](https://remix.run/stacks).

So basically what I think I would want once this PR is merged is getting the same `Dockerfile` as we did before with the `remix`/`remix_prisma` templates

Reply:
Something like the following should work I think

```Docker
# base node image
FROM node:{{ .nodeVersion }}-bullseye-slim as base
```

Reply:
Thinking more about this: should we only set major version here instead of something like `16.15.1`? 🤔

Reply:
curl is needed by node-build.

The question to ponder: do any remix applications make use of any server side npm modules that may need to be compiled using node-gyp?  If so, the remaining three are required.  Not including them will lead to a bad first experience when deploying such applications on fly.io.  Also, these packages are only installed during the build phase and are not present on the deployed machine.

the next release of debian is likely to come out this summer, and after that point new versions will likely be based on bookworm.  Do we really want people's first experience with a new version of node to be in production?

Reply:
> `curl` is needed by `node-build`.

But we only need `node-build` because we're building Node ourselves.
Using an already existing Node image (by doing `node:{{ .nodeVersion }}-bullseye-slim`) would have the same result without extra code in the `Dockerfile`

> do any remix applications make use of any server side npm modules that may need to be compiled using node-gyp? If so, the remaining three are required

The current Remix templates just use 

```Docker
FROM node:16-bullseye-slim as base
```

and it just works fine for most use-cases.

> Not including them will lead to a bad first experience when deploying such applications on fly.io

People own the `Dockerfile` themselves, so if people have special needs, they can always include `python-is-python3`, `pkg-config` & `build-essential` themselves.
I think the majority of people won't need it, so leaving them out by default wouldn't be a problem.

If people really need it (more than expected), we can always add something extra to the scanner and add them conditionally, just like we do with `openssl` for Prisma 🤷‍♂️

> these packages are only installed during the build phase and are not present on the deployed machine.

That's exactly my point: I don't think we need to build Node ourselves & we can just use the existing Node images

> Do we really want people's first experience with a new version of node to be in production?

The current `remix`/`remix_prisma` templates are using `bullseye`, but I think we can just use `node:{{ .nodeVersion }}-slim` as well (I'm not that much into Docker though)

---

> Thinking more about this: should we only set major version here instead of something like `16.15.1`? 🤔

Some other thoughts I have with how the current Node template is implemented & my question above: the used version is _really_ specific, so if we both have another Node version (for instance: I have `16.2.0` and you have `16.15.1`) we would both get a different base image to start with
Using a major version would make the chance of different `Dockerfile`s much less.

Could we maybe also look if we have an `.nvmrc` file & use that version first? 🤔
An `.nvmrc` file is most of the time comitted, so even if we have a patch version in there, we would both get the same `Dockerfile` if we would create one with `fly launch`

Reply:
you are correct that we don't need node-build (and therefore not curl).   Yes, `node:{{ .nodeVersion }}-slim` would work fine.

And yes `.nvmrc` files should be respected.  And [.node-version](https://stackoverflow.com/questions/27425852/what-uses-respects-the-node-version-file/), and [.tool-versions](https://asdf-vm.com/manage/configuration.html#tool-versions) and perhaps others.

I'm still strongly opposed to people with "special needs" first experience with fly.io is to see an error that they don't understand, being told to update a file that they aren't familiar with, and become an expert on an operating system that they don't use daily.

Reply:
> And [.node-version](https://stackoverflow.com/questions/27425852/what-uses-respects-the-node-version-file/), and [.tool-versions](https://asdf-vm.com/manage/configuration.html#tool-versions) and perhaps others.

Haven't come across them, but we could definitely add support for these files as well I think
Only "problem" I see is the order of resolution if we have both `.nvmrc` & `.node-version`, but don't see that as a blocker of not supporting those other files, just something that we need to look into how other libraries are doing it so that we have an intuitive resolution here.

> I'm still strongly opposed to people with "special needs" first experience with fly.io is to see an error that they don't understand

People would already have that currently when running `fly launch` in a Remix app, but that's not a reason to keep it that way.
My main concern is adding extra things to `Dockerfile` that most people probably don't need.

If we _know_ what packages are needing them, we could maybe add them conditionally if these packages are present in `package.json`?
That way, people not having these would not have these in their `Dockerfile`